test/core/circuit-relay.spec.js

```diff
@@ -1,145 +0,0 @@
-
-  let factory
-
-  let jsRelay = new API(`/ip4/127.0.0.1/tcp/31015`)
-  // let goRelay = new API(`/ip4/127.0.0.1/tcp/33031`)
-  let node1
-  let node2
-
-  let jsRelayId
-  let goRelayId
-
-  // let nodeId1
-  let nodeId2
-
-  before(function (done) {
-    this.timeout(40 * 1000)
-
-    factory = new IPFSFactory()
-
-    const base = {
-      EXPERIMENTAL: {
-        Relay: {
-          Enabled: true
-        }
-      },
-      Addresses: {
-        Swarm: []
-      }
-    }
-
-    parallel([
-      (cb) => factory.spawnNode(null, Object.assign(base, {
-        Addresses: {
-          Swarm: [ (isNode ? `/ip4/127.0.0.1/tcp/0` : '') ]
-        }
-      }), cb),
-      (cb) => factory.spawnNode(null, Object.assign(base, {
-        Addresses: {
-          Swarm: [ (isNode ? `/ip4/127.0.0.1/tcp/0/ws` : '') ]
-        }
-      }), cb)
-    ], (err, nodes) => {
-      expect(err).to.not.exist()
-      node1 = nodes[0]
-      node2 = nodes[1]
-
-      parallel([
-        (cb) => jsRelay.id(cb),
-        // (cb) => goRelay.id(cb),
-        (cb) => node1.id(cb),
-        (cb) => node2.id(cb)
-      ], (err, res2) => {
-        expect(err).to.not.exist()
-        parallel([
-          (cb) => peerInfoFromObj(res2[0], cb),
-          // (cb) => peerInfoFromObj(res2[1], cb),
-          (cb) => peerInfoFromObj(res2[1], cb),
-          (cb) => peerInfoFromObj(res2[2], cb)
-        ], (err, res3) => {
-          expect(err).to.not.exist()
-          jsRelayId = res3[0]
-          // goRelayId = res3[1]
-          // nodeId1 = res3[2]
-          nodeId2 = res3[2]
-          done()
-        })
-      })
-    })
-  })
-
-  after((done) => factory.dismantle(done))
-
-  // TODO: 1) figure out why this test hangs randomly
-  // TODO: 2) move this test to the interop batch
-  it.skip('node1 <-> goRelay <-> node2', (done) => {
-    const data = crypto.randomBytes(128)
-
-    series([
-      (cb) => node1.swarm.connect(goRelayId, cb),
-      (cb) => setTimeout(cb, 1000),
-      (cb) => node2.swarm.connect(goRelayId, cb),
-      (cb) => setTimeout(cb, 1000),
-      (cb) => node1.swarm.connect(nodeId2, cb)
-    ], (err) => {
-      expect(err).to.not.exist()
-      waterfall([
-        (cb) => node1.files.add(data, cb),
-        (filesAdded, cb) => node2.files.cat(filesAdded[0].hash, cb),
-        (stream, cb) => stream.pipe(bl(cb))
-      ], done)
-    })
-  })
-
-  it('node1 <-> jsRelay <-> node2', (done) => {
-    const data = crypto.randomBytes(128)
-
-    series([
-      (cb) => node1.swarm.connect(jsRelayId, cb),
-      (cb) => setTimeout(cb, 1000),
-      (cb) => node2.swarm.connect(jsRelayId, cb),
-      (cb) => setTimeout(cb, 1000),
-      (cb) => node1.swarm.connect(nodeId2, cb)
-    ], (err) => {
-      expect(err).to.not.exist()
-
-      waterfall([
-        (cb) => node1.files.add(data, cb),
-        (filesAdded, cb) => node2.files.cat(filesAdded[0].hash, cb),
-        (stream, cb) => stream.pipe(bl(cb))
-      ], done)
-    })
-  })
-})
```

Comment:
@dryajov why delete the core circuit relay tests?

Reply:
from ICR:

> 10:37 <@daviddias> dryajov: why delete the circuit relay tests?
10:38 <dryajov> moved to interop
10:39 <dryajov> they’re all there, but having them in js-ipfs in the first place was a workaround to get at least some of them running in the browser
10:39 <dryajov> now that we have the new ipfsd-ctl, thats not required anymore
10:40 <dryajov> I think we might need some around enabling/disabling circuit, but as far as those specific tests, they’re not part of interop
10:40 <dryajov> > they’re not part of interop
10:40 <dryajov> *now

Reply:
We talked about this in IRC and agreed that we need Circuit relay tests in js-ipfs to test the interface, but they will be brought back with #1063